x-pack/metricbeat/module/oracle/connection.go

```diff
@@ -53,25 +52,6 @@ func HostParser(mod mb.Module, rawURL string) (mb.HostData, error) {
 	}, nil
 }
 
-func init() {
-	// Register the ModuleFactory function for the "oracle" module.
-	if err := mb.Registry.AddModule("oracle", newModule); err != nil {
-		panic(err)
-	}
-}
-
-// newModule adds validation that hosts is non-empty, a requirement to use the
-// Oracle module.
-func newModule(base mb.BaseModule) (mb.Module, error) {
-	// Validate that at least one host has been specified.
-	config := ConnectionDetails{}
-	if err := base.UnpackConfig(&config); err != nil {
-		return nil, fmt.Errorf("error parsing config module: %w", err)
-	}
-
-	return &base, nil
-}
-
 func NewConnection(connString string) (*sql.DB, error) {
 	db, err := sql.Open("godror", connString)
```

Comment:
This still relies on godor ?

Reply:
Yes. If you check `x-pack/metricbeat/module/oracle/oracle.go`, you will notice build tags at the top. The Oracle module is now only included when CGO is enabled; otherwise, it is excluded.

However, I am also thinking of SQL module. When CGO_ENABLED=0, the SQL module in `x-pack/metricbeat` is still available, and from there, queries to an Oracle database can be sent.

Reply:
Having the tag in oracle.go ensures that module registeration doesn't happen. But if we don't isolate this function, this file will still compile for CGO disabled ?



> However, I am also thinking of SQL module

That part is handled in this PR, right ?